src/everest/detached/jobs/everserver.py

```diff
@@ -89,22 +104,50 @@ def run(self):
             )
 
         try:
-            run_model.run_experiment(evaluator_server_config)
-
+            loop = asyncio.get_running_loop()
+            simulation_future = loop.run_in_executor(
+                None,
+                lambda: run_model.run_experiment(evaluator_server_config),
+            )
+            while True:
+                if self.shared_data[STOP_ENDPOINT]:
+                    run_model.cancel()
+                    raise ValueError("Optimization aborted")
+                try:
+                    item: StatusEvents = status_queue.get(block=False)
+                except queue.Empty:
+                    await asyncio.sleep(0.01)
+                    continue
+
+                self.shared_data["events"].append(item)
+                for sub in self.shared_data["subscribers"].values():
+                    sub.notify()
+
+                if isinstance(item, EndEvent):
+                    break
+                await asyncio.sleep(0.1)
```

Comment:
what is this sleep needed for?

Reply:
To give some time between sending of events, but should be ok to remove it.